public/app/features/canvas/elements/button.tsx

```diff
@@ -132,12 +153,38 @@ export const buttonItem: CanvasElementItem<ButtonConfig, ButtonData> = {
         name: 'Style',
         editor: ButtonStyleEditor,
       })
-      .addCustomEditor({
+      .addNumberInput({
+        category,
+        path: 'config.borderWidth',
+        name: 'Button border width',
+        settings: {
+          placeholder: '0',
+        },
+      });
+    if (ctx.options?.config?.borderWidth) {
+      builder.addCustomEditor({
         category,
-        id: 'apiSelector',
-        path: 'config.api',
-        name: 'API',
-        editor: APIEditor,
+        id: 'config.borderColor',
+        path: 'config.borderColor',
+        name: 'Button border color',
+        editor: ColorDimensionEditor,
+        settings: {},
+        defaultValue: {},
       });
+    }
+    builder.addCustomEditor({
+      category,
+      id: 'apiSelector',
+      path: 'config.api',
+      name: 'API',
+      editor: APIEditor,
+    });
   },
 };
+
+const getStyles = stylesFactory((theme: GrafanaTheme2, data) => ({
+  container: css`
+    background-color: ${data?.backgroundColor};
+    border: ${data?.borderWidth}px solid ${data?.borderColor};
+  `,
```

Comment:
I think we want to avoid setting background color as that might conflict with setting the button variant color?

changing this to object notation will also fix betterer issue

```suggestion
  container: css({
    border: `${data?.borderWidth}px solid ${data?.borderColor}`,
  }),
```